pkg/cluster/cluster.go

```diff
@@ -0,0 +1,25 @@
+package cluster
+
+import (
+	"context"
+	"encoding/json"
+)
+
+// Node is the service endpoint in K8s, it's maybe podIP:port or CLUSTER-IP:port
+type Node struct {
+	// Cluster k8s' namespace
+	Namespace string
+	// Pod's name
+	PodName string
+
+	IP   string
+	Port string
+}
+
+// Provisioner provides a collection of APIs to deploy/destroy a cluster
+type Provisioner interface {
+	// SetUp sets up cluster, returns err or all nodes info
+	SetUp(ctx context.Context, spec json.RawMessage) (error, []Node)
```

Comment:
Will this interface change in the future, compare with RawMessage input, I prefer a struct or an interface.

Reply:
yep, you're right. I'll change it.